src/main/java/org/sagebionetworks/bridge/hibernate/HibernateTemplateDao.java

```
@@ -0,0 +1,106 @@
+                TEMPLATE_TYPE, type, 
+                "studyId", studyId.getIdentifier());
+        String countQuery = SELECT_COUNT + ((!includeDeleted) ? GET_ACTIVE : GET_ALL);
+        String getQuery = SELECT_TEMPLATE + ((!includeDeleted) ? GET_ACTIVE : GET_ALL);
+        
+        int total = hibernateHelper.queryCount(countQuery, params);
+        List<? extends HibernateTemplate> results = hibernateHelper.queryGet(
+                getQuery, params, offset, pageSize, HibernateTemplate.class);
+        
+        return new PagedResourceList<>(results, total)
+                .withRequestParam(TEMPLATE_TYPE, type)
+                .withRequestParam(PAGE_SIZE, pageSize)
+                .withRequestParam(OFFSET_BY, offset)
+                .withRequestParam(TOTAL, total)
+                .withRequestParam(INCLUDE_DELETED, includeDeleted);
+    }
+    
+    @Override
+    public Optional<Template> getTemplate(StudyIdentifier studyId, String guid) { 
+        checkNotNull(guid);
+        
+        Template template = hibernateHelper.getById(HibernateTemplate.class, guid);
+        if (template != null && !template.getStudyId().equals(studyId.getIdentifier())) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(template);
+    }
+
+    @Override
+    public void createTemplate(Template template) {
+        checkNotNull(template);
+        
+        hibernateHelper.create(template, null); 
+    }
+
+    @Override
+    public void updateTemplate(Template template) {
+        checkNotNull(template);
+
+        hibernateHelper.update(template, null); 
+    }
+
+    @Override
+    public void deleteTemplatePermanently(StudyIdentifier studyId, String guid) {
+        checkNotNull(guid);
+        
+        Optional<Template> template = getTemplate(studyId, guid);
+        if (template.isPresent()) {
+            hibernateHelper.deleteById(HibernateTemplate.class, guid);    
+        }
```

Unit test for deleting a template that doesn't exist?

---

In a rush I didn't look at the design document for this... in fact we're supposed to throw a ENFE if the template does not exist, but that behavior is in the service... I'm changing the service and adding a test in the DAO that it silently does nothing (in practice this should not happen if you access through the service).